gateway/api/views.py

```diff
@@ -250,59 +233,81 @@ def run(self, request):
                 )
-            except InternalServerErrorException as exception:
-                return Response(exception, exception.http_code)
+            if not program_serializer.is_valid():
+                logger.error(
+                    "RunProgramModelSerializer validation failed with program instance:\n %s",
+                    program_serializer.errors,
+                )
+                return Response(
+                    program_serializer.errors, status=status.HTTP_400_BAD_REQUEST
+                )
+            program = program_serializer.save(author=author)
 
             jobconfig = None
-            config_data = request.data.get("config")
-            if config_data:
-                config_serializer = self.get_serializer_job_config(
-                    data=json.loads(config_data)
-                )
-                if not config_serializer.is_valid():
-                    return Response(
-                        config_serializer.errors, status=status.HTTP_400_BAD_REQUEST
+            config_json = serializer.data.get("config")
+            if config_json:
+                logger.info("Configuration for [%s] was found.", title)
+                job_config_serializer = self.get_serializer_job_config(data=config_json)
+                if not job_config_serializer.is_valid():
+                    logger.error(
+                        "JobConfigSerializer validation failed:\n %s",
+                        serializer.errors,
                     )
-                try:
-                    jobconfig = (
-                        self.get_service_job_config_class().save_with_serializer(
-                            config_serializer
-                        )
+                    return Response(
+                        job_config_serializer.errors, status=status.HTTP_400_BAD_REQUEST
                     )
-                except InternalServerErrorException as exception:
-                    return Response(exception, exception.http_code)
+                jobconfig = job_config_serializer.save()
+                logger.info("JobConfig [%s] created.", jobconfig.id)
 
-            job = None
             carrier = {}
             TraceContextTextMapPropagator().inject(carrier)
-            arguments = serializer.data.get("arguments") or "{}"
-            token = request.auth.token.decode()
-            try:
-                job = self.get_service_job_class().save(
-                    program=program,
-                    arguments=arguments,
-                    author=author,
-                    jobconfig=jobconfig,
-                    token=token,
-                    carrier=carrier,
+            arguments = serializer.data.get("arguments")
+            token = ""
+            if request.auth:
+                token = request.auth.token.decode()
+            job_serializer = self.get_serializer_run_and_run_existing_job(data={})
+            if not job_serializer.is_valid():
```

Comment:
Why is this creating the empty serializer and validating it?

Reply:
That's a good question. I'm not convinced with what I did there. My assumption was like, I had some problems creating the serializer for the `Job` because the fields where relatively custom to validate:
```
job_serializer.save(
    arguments=arguments,
    author=author,
    carrier=carrier,
    token=token,
    program=program,
    config=jobconfig,
)
``` 
And the `is_valid` is mandatory if I want to call the `save` method. I accept new ideas to improve the approach 😅

Reply:
Can you put job object like
```
job_serializer = self.get_serializer_run_and_run_existing_job(
    data = job(
        arguments=arguments,
        author=author,
        carrier=carrier,
        token=token,
        program=program,
        config=jobconfig,
    )
)
if not job_serializer.is_valid():
...
```
then save it without additional fields?
```
 job = job_serializer.save()
```

Reply:
I couldn't do that due to what I explained here: https://github.com/Qiskit-Extensions/quantum-serverless/pull/1286#discussion_r1567511446

`carrier` and `token` are fields that are not in the model so I would need to create two serializers if I want to use the serializer for those. What I could maybe do is the next one:
```python
job_serializer = self.get_serializer_run_and_run_existing_job(data={
        arguments=arguments,
        program=program,
        config=jobconfig,
})
if job_serializer.is_valid():
...
job = job_serializer.save(
  author=author,
  carrier=carrier,
  token=token,
)
```

And create a specific `JobSerializer` for this case. I remember that I tried it but I had some problems configuring it. I can do a branch on top of it just exploring how to improve. I will share it here.

Reply:
Let me know what you think about the changes for this review, Aki: [8184378](https://github.com/Qiskit-Extensions/quantum-serverless/pull/1286/commits/81843781389b9940b9a4ef73838bbb83f61aaa63)

Reply:
It looks much better.  Thanks!